packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -103,11 +123,23 @@ export const makeInnerVault = (
   const { zcfSeat: liquidationZcfSeat, userSeat: liquidationSeat } =
     zcf.makeEmptySeatKit(undefined);
 
-  /** @type {VAULT_STATE} */
-  let vaultState = VaultState.ACTIVE;
+  /** @type {VAULT_PHASE} */
+  let vaultPhase = VaultPhase.ACTIVE;
 
   const assertVaultIsOpen = () => {
-    assert(vaultState === VaultState.ACTIVE, X`vault must still be active`);
+    assert(vaultPhase === VaultPhase.ACTIVE, X`vault must still be active`);
+  };
+
+  /**
+   * @param {VAULT_PHASE} newPhase
+   */
+  const assignPhase = newPhase => {
```

Comment:
If we want a state machine...then we have a library for it, and can potentially integrate that with the `enum` proposal

Reply:
Nice. I was considering a fuller state machine with transition functions and figured this was a good incremental step that wasn't entirely out of scope for the ticket :)

I'd like to hear more about the library we have and this `enum` proposal and consider refactoring this in the future to use them.

Reply:
It's not much: `.../zoe/src/contractSupport/stateMachine.js`. It started approximately like this. Just enough for present purposes, with a hope for more when we need it.